swap/src/cli/command.rs

```diff
@@ -208,3 +437,484 @@ fn parse_monero_address(s: &str) -> Result<monero::Address> {
+                    force: false,
+                    bitcoin_electrum_rpc_url: Url::from_str(DEFAULT_ELECTRUM_RPC_URL_TESTNET)
+                        .unwrap(),
+                    bitcoin_target_block: DEFAULT_BITCOIN_CONFIRMATION_TARGET_TESTNET,
+                },
+            }
+        }
+
+        pub fn cancel_mainnet_defaults() -> Self {
+            Self {
+                env_config: env::Mainnet::get_config(),
+                debug: false,
+                data_dir: data_dir_path_cli().join(MAINNET),
+                cmd: Command::Cancel {
+                    swap_id: Uuid::from_str(SWAP_ID).unwrap(),
+                    force: false,
+                    bitcoin_electrum_rpc_url: Url::from_str(DEFAULT_ELECTRUM_RPC_URL).unwrap(),
+                    bitcoin_target_block: DEFAULT_BITCOIN_CONFIRMATION_TARGET,
+                },
+            }
+        }
+
+        pub fn refund_testnet_defaults() -> Self {
+            Self {
+                env_config: env::Testnet::get_config(),
+                debug: false,
+                data_dir: data_dir_path_cli().join(TESTNET),
+                cmd: Command::Refund {
+                    swap_id: Uuid::from_str(SWAP_ID).unwrap(),
+                    force: false,
+                    bitcoin_electrum_rpc_url: Url::from_str(DEFAULT_ELECTRUM_RPC_URL_TESTNET)
+                        .unwrap(),
+                    bitcoin_target_block: DEFAULT_BITCOIN_CONFIRMATION_TARGET_TESTNET,
+                },
+            }
+        }
+
+        pub fn refund_mainnet_defaults() -> Self {
+            Self {
+                env_config: env::Mainnet::get_config(),
+                debug: false,
+                data_dir: data_dir_path_cli().join(MAINNET),
+                cmd: Command::Refund {
+                    swap_id: Uuid::from_str(SWAP_ID).unwrap(),
+                    force: false,
+                    bitcoin_electrum_rpc_url: Url::from_str(DEFAULT_ELECTRUM_RPC_URL).unwrap(),
+                    bitcoin_target_block: DEFAULT_BITCOIN_CONFIRMATION_TARGET,
+                },
+            }
+        }
+
+        pub fn with_data_dir(mut self, data_dir: PathBuf) -> Self {
+            self.data_dir = data_dir;
+            self
+        }
+
+        pub fn with_debug(mut self) -> Self {
+            self.debug = true;
+            self
+        }
+    }
+
+    fn data_dir_path_cli() -> PathBuf {
+        system_data_dir().unwrap().join("cli")
+    }
+
+    #[test]
+    fn given_buy_xmr_on_mainnet_then_defaults_to_mainnet() {
+        let raw_ars = vec![
+            BINARY_NAME,
+            "buy-xmr",
+            "--receive-address",
+            MONERO_MAINNET_ADDRESS,
+            "--seller-addr",
+            MUTLI_ADDRESS,
+            "--seller-peer-id",
+            PEER_ID,
+        ];
+
+        let expected_args = Arguments::buy_xmr_mainnet_defaults();
+        let args = parse_args_and_apply_defaults(raw_ars).unwrap();
+        assert_eq!(expected_args, args);
```

Comment:
Style nit: the following structure would follow "arrange, act, assert" more closely:

```suggestion
        let raw_ars = vec![
            BINARY_NAME,
            "buy-xmr",
            "--receive-address",
            MONERO_MAINNET_ADDRESS,
            "--seller-addr",
            MUTLI_ADDRESS,
            "--seller-peer-id",
            PEER_ID,
        ];

        let args = parse_args_and_apply_defaults(raw_ars).unwrap();

        assert_eq!(args, Arguments::buy_xmr_mainnet_defaults());
```

Reply:
It might be even nicer if we don't `.unwrap()` but assert against `Ok()`:

```rust
let args = parse_args_and_apply_defaults(raw_ars);

assert_eq!(args, Ok(Arguments::buy_xmr_mainnet_defaults());
```

That way, the only thing that can fail in this test is the assertion :)

Reply:
Could not easily make that happen:

```
binary operation `==` cannot be applied to type `std::result::Result<cli::command::Arguments, anyhow::Error>`
```

did not try much more though ;)

Reply:
Ah yeah, good old anyhow not being PartialEq. Well, worth the attempt I guess :)